src/main/java/frc/robot/subsystems/BallPath/Shooter/PIDShooterTrackingImpl.java

```diff
@@ -0,0 +1,363 @@
+
+package frc.robot.subsystems.BallPath.Shooter;
+
+import java.nio.file.spi.FileSystemProvider;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.concurrent.TimeUnit;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.fasterxml.jackson.databind.ser.std.StdArraySerializers.FloatArraySerializer;
+
+import ca.team3161.lib.robot.LifecycleEvent;
+import ca.team3161.lib.robot.subsystem.RepeatingIndependentSubsystem;
+import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
+import ca.team3161.lib.utils.Utils;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+
+
+import ca.team3161.lib.robot.subsystem.RepeatingIndependentSubsystem;
+
+public class PIDShooterTrackingImpl extends RepeatingIndependentSubsystem implements Shooter {
+    
+    private final TalonSRX turretMotor;
+    private final TalonFX shooterMotor;
+    private final TalonSRX hoodMotor;
+
+    private double kp = 0.0023; // 0.00175
+    private double ki = 0.00002; // 0.00002
+    private double kd = 0.00002; // 0.00002
```

Comment:
These constants should always match the non-tracking `PIDShooterImpl`'s I think, so it would be good to share them across the two files. A simple way to do this would be to make the variables in `PIDShooterImpl` as `static final double` instead of `private double`, then here in the `PIDShooterTrackingImpl` you can access them as `PIDShooterImpl.kp` for example.